base/threadfuncs.jl

```diff
@@ -0,0 +1,54 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+"""
+    Threads.foreach(f, channel::Channel; ntasks=Threads.nthreads(),
+                    schedule=:fair)
+
+Similar to `foreach(f, channel)`, but iteration over `channel` and calls to
+`f` are split across `ntasks` tasks spawned by `Threads.@spawn`. This function
+will wait for all internally spawned tasks to complete before returning.
+
+If `schedule` is `:fair`, `Threads.foreach` will spawn a new task for each work
+item, allowing Julia's scheduler to more freely load-balance work items across
+threads. Technically, in `:fair` mode, `Threads.foreach` may execute up to
+`2 * ntasks` tasks concurrently.
```

Comment:
I think it'd be better to treat one-task-per-item as an implementation detail.  It may be possible to get this without the re-spawn trick at some point #35688.  Maybe just say `:fair` tries to load-balance but it comes with somewhat high-cost per item? I think it also makes sense to explain it as implementation detail to give the users some idea about the overhead.

Reply:
Edited the docstring to focus more on the tradeoffs and less on implementation details 👍